src/services/expenses.service.js

```diff
@@ -0,0 +1,81 @@
+const { Expense } = require('../models/Expense.model');
+const { generateRandomId } = require('../Random');
+
+const getAllExpenses = async () => {
+  return Expense.findAll();
+};
+
+const getById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (expenses, id) => {
+  return expenses.filter((exp) => exp.userId === parseInt(id));
+};
+
+const filterExpensesByCategory = (expenses, category) => {
+  return expenses.filter((exp) => category.includes(exp.category));
```

Comment:
The `filterExpensesByCategory` function assumes that `category` is an array. Ensure that the input `category` is always an array before calling this function, or modify the function to handle cases where `category` is not an array.